generator/text_pat_generators.py

```diff
@@ -0,0 +1,55 @@
+import random
+import string
+from scipy.stats import geom
+
+class uniform_generator:
+    def generate(text_len, pat_len, alph = string.ascii_lowercase):
+        text = ''.join(random.choice(alph) for i in range(text_len))
+        pat = ''.join(random.choice(alph) for i in range(pat_len))
+        return (text, pat)
+
+class geometric_generator:
+    def generate(text_len, pat_len, alph = string.ascii_lowercase, p = 0.5):
+        choice = []
+        for i, val in enumerate(alph):
+            choice += [val for j in range(int(1000*geom.pmf(i+1,p)))]
```

Comment:
Well, shouldn't be that more pythonic?
choice = [v for _ in range(int(1000 * geom.pmf(i+1, p))) for i, v in enumerate(A)]